integrationtests/src/test/java/io/kroxylicious/proxy/filter/RequestResponseMarkingFilter.java

```diff
@@ -58,11 +59,14 @@ public CompletionStage<ApiMessage> apply(KrpcFilterContext context, ApiMessage b
             @Override
             public CompletionStage<ApiMessage> apply(KrpcFilterContext context, ApiMessage body) {
                 CompletableFuture<ApiMessage> result = new CompletableFuture<>();
-                try (var executor = Executors.newScheduledThreadPool(1)) {
-                    var delay = (long) (Math.random() * 200);
+                ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
```

Comment:
Does the change from try-with-resources to try finally actually make a difference?

Reply:
yes, in finally we are calling `shutdown`:

```
Initiates an orderly shutdown in which previously submitted tasks are executed, but no new tasks will be accepted. Invocation has no additional effect if already shut down.
This method does not wait for previously submitted tasks to complete execution. Use awaitTermination to do that.
```

where close does:

```
     * Initiates an orderly shutdown in which previously submitted tasks are
     * executed, but no new tasks will be accepted. This method waits until all
     * tasks have completed execution and the executor has terminated.
```

Reply:
Good catch.